lib/cartopy/tests/mpl/test_gridliner.py

```diff
@@ -497,7 +498,10 @@ def test_gridliner_title_adjust():
     plt.rcParams['axes.titley'] = None
 
     fig = plt.figure(layout='constrained')
-    fig.get_layout_engine().set(h_pad=1/8)
+    if _MPL_36:
+        fig.get_layout_engine().set(h_pad=1/8)
```

Comment:
The deprecation note on `set_constrained_layout_pads` suggest that the above should work from v3.6, but I have not actually tested.
https://matplotlib.org/stable/api/figure_api.html#matplotlib.figure.Figure.set_constrained_layout_pads